pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -278,6 +281,8 @@ type ReconcileClusterDeployment struct {
 	validateCredentialsForClusterDeployment func(client.Client, *hivev1.ClusterDeployment, log.FieldLogger) (bool, error)
 
 	protectedDelete bool
+
+	httpProxy, httpsProxy, noProxy string
```

Comment:
not a big fan of storing the proxy information in the reconciler. Go binaries automatically support reading these env vars and using them for http client requests, so all reconciler calls will be covered.
and only thing we need to do it copy these env vars to other pods we create.

So i would like to see that, us copying env vars to pods we created and not using reconciler fields.

Reply:
This was for testability reasons and we'd talked in the past about avoiding reading env vars directly in controllers, but I could live with the controllers doing Getenv, skip testing at that level, and pass the strings into SetProxyEnvVars which can still be unit tested cleanly. I'll make the change.